code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -216,14 +229,20 @@
 	var/mob/living/C = owner
 	var/mob/camera/aiEye/remote/xenobio/remote_eye = C.remote_control
 	var/obj/machinery/computer/camera_advanced/xenobio/X = target
+	var/obj/machinery/monkey_recycler/recycler = X.connected_recycler
 
+	if(!recycler)
+		to_chat(owner, "<span class='notice'>There is no connected monkey recycler.  Use a multitool to link one.</span>")
+		return
 	if(GLOB.cameranet.checkTurfVis(remote_eye.loc))
 		for(var/mob/living/carbon/monkey/M in remote_eye.loc)
 			if(M.stat)
 				M.visible_message("[M] vanishes as [M.p_theyre()] reclaimed for recycling!")
-				X.monkeys = round(X.monkeys + 0.2,0.1)
-				to_chat(owner, "[X] now has [X.monkeys] monkeys available.")
+				recycler.use_power(500)
+				X.monkeys += recycler.cube_production
+				X.monkeys = FLOOR(X.monkeys, 0.1)		//Prevents rounding errors	
```

Comment:
Similar to above, I'm not seeing the crazy math that requires this safeguard.

Reply:
Same as above